### ISSUE TITLE
c++20 is incorrectly used as the default cppStandard with g++-10

### ISSUE THREAD

**Issue by selectstriker2** [microsoft/vscode-cpptools#5536]
**Type: LanguageService**

**Can't use `module` for a variable name when C++ Standard has not been selected**
- OS and Version: Windows 10 10.0.17763
- VS Code Version: 1.45.1
- C/C++ Extension Version: 0.28.0
- Single file in workspace

**Steps to reproduce**

module.cpp
```
class Module
{
private:
    int module;
public:
    Module(int module);
    ~Module();
};

Module::Module(int module)
{
    //module expects a ')'
}
```

c_cpp_properties.json
```
{
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ]
        }
    ],
    "version": 4
}
```

**Expected behavior**
Expect to variable names of `module` to not be identified as errors

<details>
  <summary><strong>Logs</strong></summary>

```
-------- Diagnostics - 5/20/2020, 10:21:28 AM
Version: 0.28.0
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "compilerPath": "C:\\msys64\\mingw32\\bin",
    "compilerArgs": [],
    "intelliSenseMode": "gcc-x86",
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Keny\Projects\VSCode Issues\module.cpp ]:
    C:\KENY\PROJECTS\VSCODE ISSUES\MODULE.CPP
Translation Unit Configurations:
[ C:\Keny\Projects\VSCode Issues\module.cpp ]:
    Process ID: 10264
    Memory Usage: 15 MB
    Compiler Path: C:\msys64\mingw32\bin\gcc.exe
    Includes:
        C:\MSYS64\MINGW32\INCLUDE\C++\10.1.0
        C:\MSYS64\MINGW32\INCLUDE\C++\10.1.0\I686-W64-MINGW32
        C:\MSYS64\MINGW32\INCLUDE\C++\10.1.0\BACKWARD
        C:\MSYS64\MINGW32\LIB\GCC\I686-W64-MINGW32\10.1.0\INCLUDE
        C:\MSYS64\MINGW32\INCLUDE
        C:\MSYS64\MINGW32\LIB\GCC\I686-W64-MINGW32\10.1.0\INCLUDE-FIXED
        C:\MSYS64\MINGW32\I686-W64-MINGW32\INCLUDE
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: c++20
    IntelliSense Mode: gcc-x86
    Other Flags:
        --g++
        --gnu_version=100100
Total Memory Usage: 15 MB
```
</details>

**Comment by sean-mcmanus** [comment]
There are 2 issues:
1. We're incorrectly setting the default to c++20 for g++-10 instead of gnu++14.
2. There's an IntelliSense bug with c++20 mode in regards to the usage of "modules". I'll move that to a new bug.

The workaround is to set the cppStandard to a version less than c++20.

c18 is also used as the default for .c files, which is also incorrect (should be gnu11).

**Comment by sean-mcmanus** [comment]
Actually, I believe the usage of c++20 is "by design" because we try to use the highest standard version the compiler supports, but we may want to change that design to either use the actual default or excluding c++20 since the support is not complete yet.

Setting C_Cpp.default.cppStandard should be a workaround.

**Comment by ghuser404** [comment]
@sean-mcmanus Everyone expects IntelliSense to match compiler behaviour. Using actual default will make more sense.

**Comment by sean-mcmanus** [comment]
This seems fixed already in 0.28.3. Let me know if you can still repro it.
